api/metrics/prometheus.go

```diff
@@ -0,0 +1,75 @@
+// Package metrics provides functionality to instrument your API servers
+// using metrics such as Prometheus.
+package metrics
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/redpanda-data/common-go/api/interceptor"
+)
+
+// Prometheus can be used to instrument your API server and collect
+// a standard set of metrics on the server. It is designed to work
+// alongside the interceptor.Observer middleware.
+type Prometheus struct {
+	registry         prometheus.Registerer
+	constLabels      prometheus.Labels
+	metricsNamespace string
```

Comment:
can we add an option for metricsNamespace ?

Reply:
Ha good find, I missed this one!